classes/issue/Collector.inc.php

```diff
@@ -328,13 +350,26 @@ public function getQueryBuilder(): Builder
                     ->from('issues as i')
                     ->leftJoin('dois as d', 'd.doi_id', '=', 'i.doi_id')
                     ->whereIn('d.status', $this->doiStatuses);
+            });
+        });
 
-                $q->when(
-                    (in_array(Doi::STATUS_UNREGISTERED, $this->doiStatuses) && !$this->strictDoiStatusFilter),
-                    function (Builder $q) {
-                        $q->orWhereNull('d.status');
-                    }
-                );
+        // By whether issue has DOI assigned
+        $q->when($this->hasDois !== null, function (Builder $q) {
+            $q->whereIn('i.issue_id', function (Builder $q) {
+                $q->select('current_i.issue_id')
+                    ->from('issues', 'current_i')
+                    ->where(function (Builder $q) {
+                        $q->when($this->hasDois === true, function (Builder $q) {
+                            $q->when(in_array(Repo::doi()::TYPE_ISSUE, $this->enabledDoiTypes), function (Builder $q) {
+                                $q->whereNotNull('current_i.doi_id');
+                            });
+                        });
+                        $q->when($this->hasDois === false, function (Builder $q) {
+                            $q->when(in_array(Repo::doi()::TYPE_ISSUE, $this->enabledDoiTypes), function (Builder $q) {
+                                $q->whereNull('current_i.doi_id');
+                            });
+                        });
+                    });
```

Comment:
Minor cleanup: if you put the `$q->when(in_array(issue, enabledtypes))` outside of `$q->when($this->hasDois`) you can reduce the nested conditionals.